ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,226 @@
-            -   Click the 'SSL Options...' button and tick the 'Enable SSL' box, then click the 'OK' button.
-            -   ![](../images/odbc-driver-settings.png)
-        c.  Click the 'Test' button to verify the connection has worked. You should see the following message. *If you get an error here, repeat steps 5.e.i -- 5.e.ix again and ensure all the values are correct.*
+#### Cluster ID
 
-        ![](../images/databricks-test-connection.png)
+------------------------------------------------------------------------
 
-        d.  Click the 'OK' button to exit the 'Test Results' window, then the 'OK' button in the 'Simba Spark ODBC Driver DSN Setup' window.
+The Cluster ID is the cluster that you'll be using to run your code.
 
-5.  Connect through RStudio. Watch the below video and view the [ADA_RStudio_connect GitHub repo](https://github.com/dfe-analytical-services/ADA_RStudio_connect) for methods on connecting to Databricks and querying data from RStudio. 
+To obtain this, follow these steps:
 
+-   click 'Compute' in the Databricks menu, then click on your cluster's name
+-   in the address bar, copy the string of characters between `clusters` and `configuration` in the URL. This is your cluster ID.
 
 ------------------------------------------------------------------------
 
-# Pulling data into R studio from Databricks
+### Pulling data into RStudio from Databricks
 
 ------------------------------------------------------------------------
 
-Once you have set up an ODBC connection as detailed above, you can then use that connection to pull data directly from Databricks into R Studio. Charlotte recorded a video demonstrating two possible methods of how to do this. The recording is embedded below: 
+Now that you have established a connection between Databricks and your laptop, and between Databricks and RStudio, you can add code to your existing scripts to pull data into RStudio for analysis. If you have connected to other SQL databases before, this code will look quite familiar to you. 
+
+To access the data, we will make use of the `ODBC` package. You can find documentation about this package [on the Posit website](https://solutions.posit.co/connections/db/r-packages/odbc/).
+
+You can connect to your data in two ways, both of which are described below.
+
+---
+
+#### Connecting via the SQL Warehouse
+
+---
+
+This uses the `ODBC` package and can interact with tables in the Delta Lake. 
+
+Include the following code in your R Script:
+
+
+``` {r databricks_connect_sql, eval=FALSE}
+library(odbc)
+
+con <- DBI::dbConnect(
+    odbc::databricks(),
+    httpPath = Sys.getenv("DATABRICKS_SQL_WAREHOUSE_ID")
+)
+
+odbcListObjects(con)
+```
+
+---
+
+#### Connecting via a compute cluster
+
+---
+
+Compute clusters can interact with Volumes in Databricks, unlike the SQL Warehouse. For compute clusters, you need to use `sparklyr` rather than the `ODBC` package.
+
+Include the following code in your R Script:
+
```

Comment:
I forgot one of the dependencies here is having Python installed. We should add Python to the dependencies at the top, and say something here about the first it's run R will ask whether to install a virtual environment etc.

Reply:
This is exactly why I think `sparklyr` should be kept to a separate guidance page. There is so much that can go wrong with the install due to the behind the scenes complexity and numerous moving parts that just throwing it out as an after-thought at the end of the article seems like it's going to cause people more issues than be informative.

Especially considering that this odbc/DBI method works equally for SQL warehouses and personal clusters

Reply:
Agree with Nick here - I'll keep it out of this page for now!